backend/server.js

```
@@ -69,6 +70,105 @@
+            return res.type('text/plain').send(text);
+        } else if (fetchRes.body && (contentType.startsWith('video/') || contentType.startsWith('image/'))) {
+            // For video/image, stream as-is
+            res.set('content-type', contentType);
+            return fetchRes.body.pipe(res);
+        } else {
+            // For all other types, return JSON error, never HTML
+            return res.status(502).json({ error: 'Unsupported content-type from remote server', contentType });
+        }
+    } catch (e) {
+        // Always return JSON for errors
+        res.status(502).json({ error: 'Failed to fetch from remote server', details: e && e.message ? e.message : String(e) });
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)